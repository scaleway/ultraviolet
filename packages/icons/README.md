# Ultraviolet Icons

[![npm version](https://badge.fury.io/js/%40ultraviolet%2Ficons.svg)](https://badge.fury.io/js/%40ultraviolet%2Ficons)

Ultraviolet Icons is a set of components based on SVG icons. It provides a set of icons that can be used in your projects.

## Get Started

```sh
$ pnpm add @ultraviolet/icons @ultraviolet/themes
```

You can then add `ThemeProvider` to your applications and use the provided theme from `@ultraviolet/themes` or use your own.

```tsx
import { ThemeProvider } from '@emotion/react'
import { SdkGoProductIcon } from '@ultraviolet/icons/product'
import { theme } from '@ultraviolet/themes'

const App = () => (
  <ThemeProvider theme={theme}>
    <SdkGoProductIcon size="medium" variant="primary"/>
  </ThemeProvider>
)
```

> **Note**:
> To generate your own theme easily you can check the [theme generator](https://storybook.ultraviolet.scaleway.com/?path=/docs/tools-theme-generator--docs).

N.B. To allow typescript theme typings with `@emotion/styled` components,
you'll have to define the `@emotion/react` module `Theme` interface in your project.

Example, in a `emotion.d.ts` file:

```ts
import '@emotion/react'
import type { consoleLightTheme } from '@ultraviolet/themes'

type UltravioletUITheme = typeof consoleLightTheme

declare module '@emotion/react' {
  export interface Theme extends UltravioletUITheme {}
}
```

## Documentation

Checkout our [documentation website](https://storybook.ultraviolet.scaleway.com/).

## Contributing

üìù You can participate in the development and [start contributing](/CONTRIBUTING.md) to it.
