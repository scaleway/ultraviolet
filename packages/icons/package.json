{
  "name": "@ultraviolet/icons",
  "version": "4.0.0-beta.1",
  "description": "Ultraviolet Icons",
  "homepage": "https://github.com/scaleway/ultraviolet#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/scaleway/ultraviolet.git",
    "directory": "packages/icons"
  },
  "scripts": {
    "type:generate": "tsc --declaration -p tsconfig.build.json",
    "watch": "pnpm run '/^watch:.*/'",
    "watch:build": "vite build --config vite.config.ts --watch",
    "build": "vite build --config vite.config.ts && pnpm run type:generate",
    "build:profile": "npx vite-bundle-visualizer -c vite.config.ts",
    "typecheck": "tsc --noEmit",
    "size": "size-limit",
    "lintpublish": "publint"
  },
  "keywords": [
    "react",
    "reactjs",
    "ui"
  ],
  "license": "Apache-2.0",
  "publishConfig": {
    "access": "public"
  },
  "engines": {
    "node": ">=18.x",
    "pnpm": ">=9.x"
  },
  "os": [
    "darwin",
    "linux"
  ],
  "sideEffects": false,
  "type": "module",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "main": "./dist/index.cjs",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "require": "./dist/index.cjs",
      "import": "./dist/index.js",
      "default": "./dist/index.js"
    },
    "./*": {
      "types": "./dist/components/Icon/__generated__/*.d.ts",
      "require": "./dist/components/Icon/__generated__/*.cjs",
      "import": "./dist/components/Icon/__generated__/*.js",
      "default": "./dist/components/Icon/__generated__/*.js"
    },
    "./product": {
      "types": "./dist/components/ProductIcon/index.d.ts",
      "require": "./dist/components/ProductIcon/index.cjs",
      "import": "./dist/components/ProductIcon/index.js",
      "default": "./dist/components/ProductIcon/index.js"
    },
    "./product/*": {
      "types": "./dist/components/ProductIcon/__generated__/*.d.ts",
      "require": "./dist/components/ProductIcon/__generated__/*.cjs",
      "import": "./dist/components/ProductIcon/__generated__/*.js",
      "default": "./dist/components/ProductIcon/__generated__/*.js"
    },
    "./category": {
      "types": "./dist/components/CategoryIcon/index.d.ts",
      "require": "./dist/components/CategoryIcon/index.cjs",
      "import": "./dist/components/CategoryIcon/index.js",
      "default": "./dist/components/CategoryIcon/index.js"
    },
    "./category/*": {
      "types": "./dist/components/CategoryIcon/__generated__/*.d.ts",
      "require": "./dist/components/CategoryIcon/__generated__/*.cjs",
      "import": "./dist/components/CategoryIcon/__generated__/*.js",
      "default": "./dist/components/CategoryIcon/__generated__/*.js"
    },
    "./logo": {
      "types": "./dist/components/Logo/index.d.ts",
      "require": "./dist/components/Logo/index.cjs",
      "import": "./dist/components/Logo/index.js",
      "default": "./dist/components/Logo/index.js"
    },
    "./logo/*": {
      "types": "./dist/components/Logo/__generated__/*.d.ts",
      "require": "./dist/components/Logo/__generated__/*.cjs",
      "import": "./dist/components/Logo/__generated__/*.js",
      "default": "./dist/components/Logo/__generated__/*.js"
    }
  },
  "size-limit": [
    {
      "path": [
        "dist/**/*.js",
        "!*.d.ts",
        "!*.cjs"
      ],
      "limit": "600 kB",
      "webpack": false,
      "brotli": true,
      "running": false
    }
  ],
  "peerDependencies": {
    "@emotion/react": "11.14.0",
    "@emotion/styled": "11.14.0",
    "react": "18.x || 19.x",
    "react-dom": "18.x || 19.x"
  },
  "devDependencies": {
    "@babel/core": "7.26.10",
    "@emotion/react": "11.14.0",
    "@emotion/styled": "11.14.0",
    "@types/react": "19.0.12",
    "@types/react-dom": "19.0.4",
    "@utils/test": "workspace:*",
    "react": "19.0.0",
    "react-dom": "19.0.0"
  },
  "dependencies": {
    "@ultraviolet/themes": "workspace:*"
  }
}
