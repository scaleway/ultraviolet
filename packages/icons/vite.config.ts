import { resolve } from 'path'
import svgr from '@svgr/rollup'
import { defineConfig, mergeConfig } from 'vite'
import { defaultConfig } from '../../vite.config'

export default mergeConfig(defineConfig(defaultConfig), {
  plugins: [
    svgr({ memo: true, svgo: false }), // We disable svgo because we have custom configuration for it svgo.config.js
  ],
  build: {
    lib: {
      entry: {
        '': resolve(__dirname, 'src/index.ts'),
        'components/ProductIcon/index': resolve(
          __dirname,
          'src/components/ProductIcon/index.ts',
        ),
        'components/CategoryIcon/index': resolve(
          __dirname,
          'src/components/CategoryIcon/index.ts',
        ),
        'components/Logo/index': resolve(
          __dirname,
          'src/components/Logo/index.ts',
        ),
      },
    },
  },
})
