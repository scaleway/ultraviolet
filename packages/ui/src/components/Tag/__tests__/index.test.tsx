import { LockIcon } from '@ultraviolet/icons'
import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, test } from 'vitest'
import { Tag } from '..'

describe('tag', () => {
  test('renders correctly', () => shouldMatchEmotionSnapshot(<Tag>test</Tag>))

  test('renders correctly neutral', () =>
    shouldMatchEmotionSnapshot(<Tag sentiment="neutral">test</Tag>))

  test('renders correctly disabled', () =>
    shouldMatchEmotionSnapshot(<Tag disabled>test</Tag>))

  test('renders correctly colored', () =>
    shouldMatchEmotionSnapshot(<Tag sentiment="primary">test</Tag>))

  test('renders correctly with icon', () =>
    shouldMatchEmotionSnapshot(
      <Tag>
        <LockIcon />
        test
      </Tag>,
    ))

  test('renders correctly with isLoading', () =>
    shouldMatchEmotionSnapshot(<Tag isLoading>test</Tag>))

  test('renders correctly with onClose', () =>
    shouldMatchEmotionSnapshot(<Tag onClose={() => {}}>test</Tag>))

  test('renders correctly with copiable', () =>
    shouldMatchEmotionSnapshot(<Tag copiable>test</Tag>))
})
