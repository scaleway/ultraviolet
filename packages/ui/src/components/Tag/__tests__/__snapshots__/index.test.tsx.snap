// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag > renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-3 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

<div
    data-testid="testing"
  >
    <span
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        test
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Tag > renders correctly colored 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #641cb3;
  background: #f1eefc;
  border: solid 1px #f1eefc;
}

.emotion-3 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

<div
    data-testid="testing"
  >
    <span
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        test
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Tag > renders correctly disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #b5b7bd;
  background: #f3f3f4;
  border: solid 1px #e9eaeb;
  cursor: not-allowed;
}

.emotion-3 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

<div
    data-testid="testing"
  >
    <span
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        test
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Tag > renders correctly neutral 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-3 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

<div
    data-testid="testing"
  >
    <span
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        test
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Tag > renders correctly with copiable 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: inherit;
}

.emotion-0[data-container-full-width="true"] {
  width: 100%;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-2:hover,
.emotion-2:active {
  cursor: pointer;
  background: #e9eaeb;
  border-color: #92959d;
}

.emotion-2:active {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-5 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

<div
    data-testid="testing"
  >
    <div
      aria-controls=":r8:"
      aria-describedby=":r8:"
      class="emotion-0 emotion-1"
      tabindex="0"
    >
      <button
        class="emotion-2 emotion-3"
      >
        <div
          class="emotion-4 emotion-5 emotion-6"
        >
          test
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tag > renders correctly with icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-2 {
  vertical-align: middle;
  fill: currentColor;
  height: 16px;
  width: 16px;
  min-width: 16px;
  min-height: 16px;
}

.emotion-2 .fillStroke {
  stroke: currentColor;
  fill: none;
}

.emotion-2 path {
  fill: currentColor;
}

.emotion-5 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

<div
    data-testid="testing"
  >
    <span
      class="emotion-0 emotion-1"
    >
      <svg
        class="emotion-2 emotion-3"
        fill="none"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8 1C6.067 1 4.5 2.567 4.5 4.5L4.5 7C3.67157 7 3 7.67157 3 8.5V13.5C3 14.3284 3.67157 15 4.5 15H11.5C12.3284 15 13 14.3284 13 13.5V8.5C13 7.67157 12.3284 7 11.5 7V4.5C11.5 2.567 9.933 1 8 1ZM10 7V4.5C10 3.39543 9.10457 2.5 8 2.5C6.89543 2.5 6 3.39543 6 4.5V7H10Z"
          fill="black"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="emotion-4 emotion-5 emotion-6"
      >
        test
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Tag > renders correctly with isLoading 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-3 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

.emotion-5 {
  -webkit-animation: spin 0.75s linear infinite;
  animation: spin 0.75s linear infinite;
}

.emotion-7 {
  -webkit-transition: stroke-dashoffset 0.5s ease 0s;
  transition: stroke-dashoffset 0.5s ease 0s;
}

<div
    data-testid="testing"
  >
    <span
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        test
      </div>
      <svg
        aria-label="Loading"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        aria-valuetext="20%"
        class="emotion-5 emotion-6"
        role="progressbar"
        style="height: 16px; width: 16px;"
        viewBox="0 0 100 100"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="42"
          stroke="#d9dadd"
          stroke-width="16"
        />
        <circle
          class="emotion-7"
          cx="50"
          cy="50"
          fill="none"
          r="42"
          stroke="#8c40ef"
          stroke-dasharray="263.89378290154264"
          stroke-dashoffset="211.11502632123413"
          stroke-linecap="round"
          stroke-width="16"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Tag > renders correctly with onClose 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-3 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 2rem;
  padding: 0 0.5rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  paragraph-spacing: 0;
  text-case: none;
  background: none;
  border: none;
  color: #3f4250;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0.125rem;
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:active {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-6:hover,
.emotion-6:active {
  background: #e9eaeb;
  color: #222638;
}

.emotion-8 {
  vertical-align: middle;
  fill: currentColor;
  height: 16px;
  width: 16px;
  min-width: 16px;
  min-height: 16px;
}

.emotion-8 .fillStroke {
  stroke: currentColor;
  fill: none;
}

.emotion-8 path {
  fill: currentColor;
}

<div
    data-testid="testing"
  >
    <span
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3 emotion-4"
      >
        test
      </div>
      <button
        aria-label="Close tag"
        class="emotion-5 emotion-6 emotion-7"
        data-testid="close-tag"
        type="button"
      >
        <svg
          class="emotion-8 emotion-9"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;
